bug(docs): `HybridCache` doctests fail without `redis-backend` feature

## Description

4 doctests in `crates/reinhardt-utils/src/cache/hybrid.rs` fail when `redis-backend` feature is not enabled because they import `RedisCache` unconditionally.

## Error

```
error[E0432]: unresolved import `reinhardt_utils::cache::RedisCache`
```

## Failing Tests

- `cache::hybrid (line 15)`
- `cache::hybrid::HybridCache<L1,L2>::new (line 79)`
- `cache::hybrid::HybridCache<L1,L2>::l1 (line 100)`
- `cache::hybrid::HybridCache<L1,L2>::l2 (line 121)`

## Suggested Fix

Add feature gates to the doc examples:

```rust
/// ```rust,ignore
/// // or use cfg_attr:
/// #[cfg(feature = "redis-backend")]
/// ```

Or change examples to use only always-available types (e.g., `InMemoryCache` for both L1 and L2).

--- a/crates/reinhardt-utils/src/cache/hybrid.rs
+++ b/crates/reinhardt-utils/src/cache/hybrid.rs
@@ -12,14 +12,14 @@
 //!
 //! # Examples
 //!
-//! ```no_run
-//! use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache, RedisCache};
+//! ```
+//! use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache};
 //! use std::time::Duration;
 //!
 //! # async fn example() -> Result<(), Box<dyn std::error::Error>> {
-//! // Create L1 (memory) and L2 (Redis) caches
+//! // Create L1 (memory) and L2 (distributed) caches
 //! let l1_cache = InMemoryCache::new();
-//! let l2_cache = RedisCache::new("redis://localhost:6379").await?;
+//! let l2_cache = InMemoryCache::new(); // In production, use RedisCache or MemcachedCache
 //!
 //! // Create hybrid cache
 //! let cache = HybridCache::new(l1_cache, l2_cache);
@@ -76,15 +76,12 @@ where
 	///
 	/// # Examples
 	///
-	/// ```no_run
-	/// use reinhardt_utils::cache::{HybridCache, InMemoryCache, RedisCache};
+	/// ```
+	/// use reinhardt_utils::cache::{HybridCache, InMemoryCache};
 	///
-	/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
 	/// let l1 = InMemoryCache::new();
-	/// let l2 = RedisCache::new("redis://localhost:6379").await?;
+	/// let l2 = InMemoryCache::new();
 	/// let cache = HybridCache::new(l1, l2);
-	/// # Ok(())
-	/// # }
 	/// ```
 	pub fn new(l1: L1, l2: L2) -> Self {
 		Self {
@@ -97,12 +94,12 @@ where
 	///
 	/// # Examples
 	///
-	/// ```no_run
-	/// use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache, RedisCache};
+	/// ```
+	/// use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache};
 	///
 	/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
 	/// let l1 = InMemoryCache::new();
-	/// let l2 = RedisCache::new("redis://localhost:6379").await?;
+	/// let l2 = InMemoryCache::new();
 	/// let cache = HybridCache::new(l1, l2);
 	///
 	/// // Clear only L1 cache
@@ -118,12 +115,12 @@ where
 	///
 	/// # Examples
 	///
-	/// ```no_run
-	/// use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache, RedisCache};
+	/// ```
+	/// use reinhardt_utils::cache::{Cache, HybridCache, InMemoryCache};
 	///
 	/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
 	/// let l1 = InMemoryCache::new();
-	/// let l2 = RedisCache::new("redis://localhost:6379").await?;
+	/// let l2 = InMemoryCache::new();
 	/// let cache = HybridCache::new(l1, l2);
 	///
 	/// // Clear only L2 cache

